docs/artefacts/offramps.md

```diff
@@ -996,9 +996,347 @@ offramp:
+          "locationType": "multi-region",
+          "storageClass": "STANDARD",
+          "metageneration": "4",
+          "selfLink": "https://www.googleapis.com/storage/v1/b/<bucket-id>",
+          "kind": "storage#bucket",
+          "name": "<bucket-name>",
+          "timeCreated": "<time of creation>",
+          "iamConfiguration": {
+            "uniformBucketLevelAccess": {
+              "enabled": true,
+              "lockedTime": "2021-06-17T09:12:22.122Z"
+            },
+            "bucketPolicyOnly": {
+              "enabled": true,
+              "lockedTime": "2021-06-17T09:12:22.122Z"
+            }
+          },
+          "etag": "CAQ=",
+          "defaultEventBasedHold": false,
+          "retentionPolicy": {
+            "effectiveTime": "2021-03-19T09:12:22.122Z",
+            "retentionPeriod": "60",
+            "isLocked": false
+          },
+          "updated": "2021-04-13T17:05:29.404Z",
+          "satisfiesPZS": false,
+          "projectNumber": "376177678867"
+        }
+      ],
+      "kind": "storage#buckets"
+    }
+  }
+]
+
+```
+
+***Where***
+- `<project-id>` - The project id of the Google Cloud Storage project where the buckets are.
+- `<bucket-id>` - The unique identifier of the bucket.
+
+
+#### list_objects
+
+Lists all the objects in the specified bucket.
+
+**Request**:
+```js
+{
+  "command": "list_objects",
+  "bucket": "<bucket-name>"
+}
+```
+
+**Response**:
+```js
+[
+  {
+    "cmd": "list_objects",
+    "data": {
+      "items": [
+        {
+          "mediaLink": "",
+          "bucket": "<bucket-name>",
+          "id": "<object-id>",
+          "size": "1943550",
+          "crc32c": "0QXXmA==",
+          "storageClass": "STANDARD",
+          "generation": "1616145246990906",
+          "metageneration": "1",
+          "timeStorageClassUpdated": "2021-03-19T09:14:07.012Z",
+          "selfLink": "",
+          "kind": "storage#object",
+          "name": "<object-name>",
+          "md5Hash": "ucrP5b1N+6JHxn1TKavn/A==",
+          "timeCreated": "2021-03-19T09:14:07.012Z",
+          "etag": "CLrk6JqCvO8CEAE=",
+          "updated": "2021-03-19T09:14:07.012Z",
+          "retentionExpirationTime": "2021-03-19T09:15:07.012Z",
+          "contentType": "<content-type>"
+        }        
+      ],
+      "kind": "storage#objects"
+    }
+  }
+]
+
+```
+
+***Where***
+- `<bucket-name>` - The name of the Google Cloud Storage bucket where the object is.
+- `<object-id>` - The unique object identifier.
+- `<object-name>` - The name of the object in the Google Cloud Stoarge bucket.
+- `<content-type>` - The type of the object uploaded.
+
+
+#### create_bucket
+
+Creates a bucket in the project specified in the command.
+
+**Request**:
+```js
+{
+  "command": "create_bucket" , 
+  "project_id": "<project-id>", 
+  "bucket": "<bucket>"
+}
+```
+
+**Response**:
+```js
+[
+  {
+    "cmd": "create_bucket",
+    "data": {
+      "projectNumber": "376177678867",
```

Comment:
Lets make this value an argument and remove the project number value rather than use the actual project number